libsql-server/src/connection/write_proxy.rs

```diff
@@ -233,6 +241,23 @@ impl WriteProxyConnection<RpcStream> {
             None => Ok(()),
         }
     }
+
+    /// returns whether a request should be unconditionally proxied based on the current state of
+    /// the replica.
+    fn should_proxy(&self) -> bool {
+        // There primary has data
+        if let Some(primary_index) = self.primary_replication_index {
+            let last_applied = *self.applied_frame_no_receiver.borrow();
+            // if we either don't have data while the primary has, or the data we have is
+            // anterior to that of the primary when we loaded the namespace, then proxy the
+            // request to the primary
+            if last_applied.is_none() || last_applied.unwrap() < primary_index {
```

Comment:
```suggestion
            if last_applied < Some(primary_index) {
```

does this work? I think its a bit easier to read